### PR TITLE
nix: update p4 fusion openssl3

### PR DESCRIPTION
**Pull request by burmudar**
Mostly just updating shasums and checking that it still builds.
## Test plan
Tested on Mac and Linux

### CODE REVIEW

File: dev/nix/p4-fusion.nix [hunk 1]
```diff
@@ -56,21 +54,21 @@ let
...
+                url = "https://cdist2.perforce.com/perforce/r22.2/bin.macosx12x86_64/p4api-openssl3.tgz";
+                hash = "sha256-E5oK/uvRRNuEC5b8h5zQVdfZOieUgV2T8aPISEWgSY8=";
               }
           else if targetPlatform.isLinux then
             fetchzip
               {
                 name = "helix-core-api";
-                url = "https://cdist2.perforce.com/perforce/r22.2/bin.linux26x86_64/p4api-glibc2.3-openssl1.1.1.tgz";
-                hash = "sha256-JkWG4ImrTzN0UuSMelG8zsH7YRlL1mXs9lpB5GptUb4=";
+                url = "https://cdist2.perforce.com/perforce/r22.2/bin.linux26x86_64/p4api-glibc2.3-openssl3.tgz";
```
Comment (Strum355) [inline dev/nix/p4-fusion.nix:70]: Wonder whether we should use the glibc2.12 one 🤔

**Comment by Strum355** [comment]
Theres a docs section on updating hashes that would need updating with the new urls 🙂

**Comment by burmudar** [comment]
> Theres a docs section on updating hashes that would need updating with the new urls 🙂

@Strum355 Thanks - I'll update it!

File: dev/nix/p4-fusion.nix (hunk 1 above)
Comment (burmudar) [reply dev/nix/p4-fusion.nix:70]: It no like glibc2.12 🤔  - I feel I have seen this error before but 🤷🏼 
```
git -P diff
diff --git a/dev/nix/p4-fusion.nix b/dev/nix/p4-fusion.nix
index 575b9c9d3f..c7eff9f432 100644
--- a/dev/nix/p4-fusion.nix
+++ b/dev/nix/p4-fusion.nix
@@ -67,8 +67,8 @@ let
             fetchzip
               {
                 name = "helix-core-api";
-                url = "https://cdist2.perforce.com/perforce/r22.2/bin.linux26x86_64/p4api-glibc2.3-openssl3.tgz";
-                hash = "sha256-wovk38lk4cjrC6pTa9dSwmYM1RN1nzmzdl8VIwf2BLY=";
+                url = "https://cdist2.perforce.com/perforce/r22.2/bin.linux26x86_64/p4api-glibc2.12-openssl3.tgz";
+                hash = "sha256-vtzrX49MKVDwrr1NpM+HeFy9zAo2SCk/iFVIphNdp68=";
               }
           else throw "unsupported platform ${stdenv.targetPlatform.parsed.kernel.name}"
         )
```
```
nix build ".#packages.x86_64-linux.p4-fusion"

warning: Git tree '/home/william/code/sourcegraph' is dirty
error: builder for '/nix/store/4lkxw0d3dcs3afj1281l5qccaf4syz81-p4-fusion-x86_64-unknown-linux-musl.drv' failed with exit code 2;
       last 10 log lines:
       > /tmp/151411095/depot/r22.2/p4-bin/../p4/client/p4libs.cc:124: undefined reference to `sqlite3_initialize'
       > /nix/store/7avq1avrcp66asl6khibdk3r3jd5wway-x86_64-unknown-linux-musl-binutils-2.40/bin/x86_64-unknown-linux-musl-ld: /build/p4-fusion/p4-fusion/../vendor/helix-core-api/linux/lib/libclient.a(p4libs.o): in function `P4Libraries::Shutdown(int, Error*)':
       > /tmp/151411095/depot/r22.2/p4-bin/../p4/client/p4libs.cc:175: undefined reference to `sqlite3_shutdown'
       > /nix/store/7avq1avrcp66asl6khibdk3r3jd5wway-x86_64-unknown-linux-musl-binutils-2.40/bin/x86_64-unknown-linux-musl-ld: /tmp/151411095/depot/r22.2/p4-bin/../p4/client/p4libs.cc:178: undefined reference to `curl_global_cleanup'
       > /nix/store/7avq1avrcp66asl6khibdk3r3jd5wway-x86_64-unknown-linux-musl-binutils-2.40/bin/x86_64-unknown-linux-musl-ld: /build/p4-fusion/p4-fusion/../vendor/helix-core-api/linux/lib/libclient.a(p4libs.o): in function `P4Libraries::Initialize(int, Error*)':
       > /tmp/151411095/depot/r22.2/p4-bin/../p4/client/p4libs.cc:128: undefined reference to `curl_global_init_mem'
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [p4-fusion/CMakeFiles/p4-fusion.dir/build.make:426: p4-fusion/p4-fusion] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:245: p4-fusion/CMakeFiles/p4-fusion.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/4lkxw0d3dcs3afj1281l5qccaf4syz81-p4-fusion-x86_64-unknown-linux-musl.drv'.
```

Comment (Strum355) [reply dev/nix/p4-fusion.nix:70]: Might need to add sqlite3 and curl as build inputs 🤔 i think i had to do that

Comment (burmudar) [reply dev/nix/p4-fusion.nix:70]: Ah right - testing it out now

**Review by Strum355 — APPROVED** [review]
Lessa-go

**Pull request merged**